src/deny.ts

```diff
@@ -3,36 +3,37 @@ import * as core from '@actions/core'
 
 export async function getDeniedChanges(
   changes: Change[],
-  deniedPackages: string[],
-  deniedGroups: string[]
+  deniedPackages: string[] = [],
+  deniedGroups: string[] = []
 ): Promise<Change[]> {
   const changesDenied: Change[] = []
 
-  let failed = false
+  let hasDeniedPackage = false
   for (const change of changes) {
-    change.name = change.name.toLowerCase()
-    const packageUrl = change.package_url.toLowerCase().split('@')[0]
+    const [packageName, packageVersion] = change.package_url
+      .toLowerCase()
+      .split('@')
```

Comment:
Getting the package version from a purl is a little more complicated than this. The [spec](https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst) allows for additional "qualifiers" after the version, e.g.:

```
pkg:rpm/fedora/curl@7.50.3-1.fc25?arch=i386&distro=fedora-25
```

We may not need to worry about that in the deny list, but I think we should be prepared for those extra segments in the change list because they might get returned from snapshots.